lib/ruport/data/record.rb

```diff
@@ -55,10 +55,10 @@ def initialize(data,options={})
       case(data)
       when Array
         @attributes = options[:attributes] || (0...data.length).to_a
-        @data = @attributes.inject({}) { |h,a| h.merge(a => data.shift) }
+        @data = @attributes.each_with_object({}) { |a, h| h[a.freeze] = data.shift }
       when Hash
-        @data = data.dup
-        @attributes = options[:attributes] || data.keys
+        @data = data
```

Comment:
@kbrock `data` is dup'd on line 54, so I removed the extra `.dup` call.
I don't think I changed the behavior of this method, let me know if I'm wrong...